src/puppetlabs/trapperkeeper/internal.clj

```diff
@@ -44,6 +47,26 @@
    (every? keyword? (keys service-graph))
    (every? (some-fn ifn? service-graph?) (vals service-graph))))
 
+(defn inc-restart-counter!
+  "Increments the counter in a restart-file for purposes of supporting HUP behavior"
+  [restart-file-path]
+  (when (some? restart-file-path)
+    (try
+      (if (fs/exists? restart-file-path)
+        (spit restart-file-path (inc (Long/parseLong (slurp restart-file-path))))
+        (let [path-list (string/split restart-file-path #"/")
+              file-name (last path-list)
+              dir-path (string/join "/" (drop-last path-list))]
+          (when (not-empty (drop-last path-list))
+            (fs/mkdirs dir-path))
+          (spit restart-file-path "0")
+          (inc-restart-counter! restart-file-path)))
+      (catch ArithmeticException e
+        (spit restart-file-path "0")
+        (log/warn "Number of restarts has exceeded Long/MAX_VALUE, resetting file to 0"))
+      (catch NumberFormatException e
+        (log/error "Invalid counter; restart file must only consist of one number with no whitespace")))))
```

Comment:
`finally` will *always* run, regardless of whether an exception was thrown, so I don't think you want to use that here. https://clojuredocs.org/clojure.core/finally